public/app/features/dashboard-scene/sharing/ExportButton/ExportAsImage.tsx

```diff
@@ -0,0 +1,294 @@
+  const [ref, { width: loadingBarWidth }] = useMeasure<HTMLDivElement>();
+
+  const dashboard = getDashboardSceneFor(model);
+
+  // Clean up object URLs when component unmounts
+  useEffect(() => {
+    return () => {
+      if (imageBlob) {
+        URL.revokeObjectURL(URL.createObjectURL(imageBlob));
+      }
+    };
+  }, [imageBlob]);
+
+  const onFormatChange = (value: ImageFormat) => {
+    setFormat(value);
+  };
+
+  const onExport = async () => {
+    setIsLoading(true);
+    setError(null);
+
+    try {
+      const result = await generateDashboardImage({
+        dashboard,
+        format,
+        scale: config.rendererDefaultImageScale || 1,
+      });
+
+      if (result.error) {
+        throw new Error(result.error);
+      }
+
+      setImageBlob(result.blob);
+      DashboardInteractions.generateDashboardImageClicked({
+        format,
+        scale: config.rendererDefaultImageScale || 1,
+        shareResource: 'dashboard',
+        success: true,
+      });
+    } catch (error) {
+      console.error('Error exporting image:', error);
+      DashboardInteractions.generateDashboardImageClicked({
+        format,
+        scale: config.rendererDefaultImageScale || 1,
+        shareResource: 'dashboard',
+        success: false,
+        error: error instanceof Error ? error.message : 'Failed to generate image',
+      });
+      setError({
+        title: t('share-modal.image.error-title', 'Failed to generate image'),
+        message: error instanceof Error ? error.message : 'Failed to generate image',
+      });
+    } finally {
+      setIsLoading(false);
+    }
+  };
+
+  const onDownload = () => {
+    if (!imageBlob) {
+      return;
+    }
+
+    const time = new Date().getTime();
+    const name = dashboard.state.title;
+    saveAs(imageBlob, `${name}-${time}.${format}`);
+
+    DashboardInteractions.downloadDashboardImageClicked({
+      format,
+      fileName: `${name}-${time}.${format}`,
+      shareResource: 'dashboard',
+    });
+  };
+
+  return (
+    <>
+      <p className={styles.info}>
+        <Trans i18nKey="share-modal.image.info-text">
+          Save this dashboard as an image file. The image will be captured at high resolution.
+        </Trans>
+      </p>
+
+      <Field label={t('share-modal.image.format-label', 'Format')}>
+        <div data-testid={selectors.components.ExportImage.formatOptions.container}>
+          <RadioButtonGroup
+            value={format}
+            onChange={onFormatChange}
+            options={[
+              {
+                label: 'PNG',
+                value: ImageFormat.PNG,
+                'data-testid': selectors.components.ExportImage.formatOptions.png,
+              },
+              {
+                label: 'JPG',
+                value: ImageFormat.JPG,
+                'data-testid': selectors.components.ExportImage.formatOptions.jpg,
+              },
+            ]}
+          />
+        </div>
+      </Field>
+
+      <RendererAlert />
```

Comment:
Maybe we should show only the alert when the image renderer is not available and not the different buttons, WDYT?